packages/next-app/src/app/stableswapsimulator/(components)/PoolParametersForm.tsx

```diff
@@ -1,13 +1,76 @@
 "use client";
 
-import { useStableSwap } from "#/contexts/StableSwapContext";
+import { useEffect, useState } from "react";
+
+import Button from "#/components/Button";
+import { Input } from "#/components/Input";
+import { AnalysisData, useStableSwap } from "#/contexts/StableSwapContext";
 
 export default function PoolParametersForm() {
-  const { initialData } = useStableSwap();
+  const { initialData, setInitialData } = useStableSwap();
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [formData, setFormData] = useState(initialData);
+
+  const inputParameters = (field: keyof AnalysisData) => {
+    const label = field.replace(/([a-z])([A-Z])/g, "$1 $2").toLowerCase();
+    if (field == "tokens") return;
+    return {
+      label: label.charAt(0).toUpperCase() + label.slice(1),
+      type: "number",
+      placeholder: `Define the initial ${label}`,
+      value: formData?.[field],
+      onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+        const value =
+          e.target.value == "" ? undefined : parseFloat(e.target.value);
+        if (value == undefined) {
+          setFormData({
+            ...formData,
+            [field]: undefined,
+          });
+          return;
+        }
+        if (value < 0) return;
+        setFormData({
+          ...formData,
+          [field]: value,
+        });
+      },
+    };
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setInitialData(formData);
+  };
+
+  useEffect(() => {
+    setIsSubmitted(
+      typeof formData?.ampFactor == "number" &&
+        typeof formData?.swapFee == "number"
```

Comment:
If you want to set it as submitted just set it to true to keep it simple. If there's a reason for this it should probably have a comment around it since it's a very non obvious choice. Also likely to generate less [WTFs] imo (https://www.reddit.com/r/programming/comments/6mecg/wtfminutethe_only_valid_measurement_of_code/) (e.g. What if one of the types is not a number? Then it's not gonna be submitted and the user is gonna be stuck in a loop here? )